artifactory/commands/buildinfo/adddependencies.go

```diff
@@ -219,3 +289,35 @@ func convertFileInfoToDependencies(files map[string]*fileutils.FileDetails) []bu
 	}
 	return buildDependencies
 }
+
+func convertSearchResultToDependencies(resultItem specutils.ResultItem) buildinfo.Dependency {
+	dependency := buildinfo.Dependency{Checksum: &buildinfo.Checksum{Md5: resultItem.Actual_Md5, Sha1: resultItem.Actual_Sha1}}
+	dependency.Id = resultItem.Name
+	return dependency
+}
+
+func searchItems(spec *spec.SpecFiles, servicesManager artifactory.ArtifactoryServicesManager) (resultReader *content.ContentReader, errorOccurred bool, err error) {
+	temp := []*content.ContentReader{}
+	defer func() {
+		for _, reader := range temp {
+			reader.Close()
+		}
+	}()
+	for i := 0; i < len(spec.Files); i++ {
```

Comment:
Unlike downloading or uploading files, Idon't think we should continue the process of collecting items, if one of the searches failed. There's just no point in doing this. As soon as an error is returned, the process hsould stop.